invenio_files_rest/__init__.py

```diff
@@ -6,7 +6,788 @@
+the version id:
+
+.. code-block:: console
+
+   $ curl -i -X DELETE \
+       http://localhost:5000/files/$B/my_file.txt?versionId=<version_id>
+
+2. Delete a whole file (creates a delete marker which makes the file
+inaccessible):
+
+.. code-block:: console
+
+   $ curl -i -X DELETE http://localhost:5000/files/$B/my_file.txt
+
+
+Access control
+--------------
+
+Invenio-Files-REST depends on `Invenio-Access
+<https://invenio-access.readthedocs.io>`_ module, to control the files access.
+
+It comes with a default permission factory implementation which can be found
+at :py:data:`invenio_files_rest.permissions.permission_factory` and can be
+customized further, by providing our custom implementation in the relevant
+config variable
+:py:data:`invenio_files_rest.config.FILES_REST_PERMISSION_FACTORY`.
+
+The module also comes with a list of predefined actions for the most common
+operations:
+
+    - location-update
+    - bucket-read
+    - bucket-read-versions
+    - bucket-update
+    - bucket-listmultiparts
+    - object-read
+    - object-read-version
+    - object-delete
+    - object-delete-version
+    - multipart-read
+    - multipart-delete
+
+
+The aforementioned actions accept parameters, and can be simply applied by
+decorating our desired function. For example, to verify that the contents of
+a bucket can be read, you should have the bucket-read permission which takes
+the bucket as the argument.
+
+.. code-block:: python
+
+    @need_permissions(
+        lambda self, bucket, versions: bucket,
+        'bucket-read',
+    )
+    def my_function():
+        pass
+
+.. note::
+
+    (This is mentioned in the code, not sure if should go here)
+    The actual file access is handled by the storage interface.
+
+See :mod:`invenio_files_rest.permissions` for extensive documentation.
+
+
+Integrity
+---------
+
+Invenio-Files-REST stores file checksums and regularly revalidates them, in
+order to verify the data integrity of all data at rest, as well as to detect
+corruption of data in transit.
+
+Whenever an operation like :code:`merge_multipartobject` the database is rolled
+back to reflect the state of the system before the action.
+
+For the computation of the checksum you can provide the desired algorithm,
+otherwise :code:`MD5` will be used.
+
+In the case of trying to remove a file, the expectation is for the
+FileInstance removal to succeed in the database so that afterwards
+it's possible to proceed with the removal of the actual file. By binding the
+actions together, in case of database operation error, you can prevent the
+system ending up with orphan files. Some of the tasks support :code:`silent`
+as an argument to prevent propagation of the errors.
+
+There is also a predefined task :code:`verify_checksum` which can be configured
+to run periodically (default is every 30 days) which iterates all files in our
+storage and validates their checksum.
+
+>>> "Get the checksum of an uploaded file for demo"
```

Comment:
@topless I think this is a little bit out of context. Maybe two lines of python code that you mentioned would be better.